src/main/scala/viper/gobra/frontend/ParseTreeTranslator.scala

```diff
@@ -0,0 +1,2348 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I agree that we should group them more reasonably, I don't think function overrides can be split into multiple files. When it comes to the visit... functions, there are three main patterns:

1.  Check with an if statement in which rule alternative we are, and then just call visit...() or visitNode[P...]() on each of the constituents to assemble the resulting PNode.
2. Call visitChildren, which returns a Vector of the rule constituents and then match this to find out which alternative is chosen and construct the resulting PNode. (equivalent to 1, just more practical in some cases)
3. call visitNodeChild(), which is just a convenience function for rules that are just "collections" of single rules.

These should be easy to maintain, since that just involves adding a new match- or if-case. ANTLR also allows you to name main alternatives, this should probably be used for some of the rules with huge if-statement chains. The main problem is that there are indeed a lot of edge cases that don't fall into these three, usually when the Go grammar and the Gobra AST aren't quite compatible, and I agree that these should be commented thoroughly to make them maintainable.

Reply:
Overall, this file became much more readable. I had the impression that I could quickly figure out what each function is supposed to do and does. Thanks a lot!